doc/Example.rst

```diff
@@ -0,0 +1,44 @@
+.. _Example:
+
+Regression Testing Example
+--------------------------
+
+:class:`baseclasses.BaseRegTest.BaseRegTest` provides a framework for creating regression tests.
+It stores data in a JSON file format during _training_ and compares test results against this stored data during _testing_.
+Here, we will go through a short example of how to integrate this into Pythons's ``unittest`` framework, specifically when used with ``testflo``.
+
+In this example, a regression test for a function called ``sampling.polynomial`` will be created.
+This function returns a ``numpy`` array of values that are spaced between a start and end point according to a polynomial distribution.
+The test function is structured as follows:
+
+.. code-block:: python
+
+    def TestSampling(unittest.TestCase):
+        def test_polynomial(self, train=False):
+            ref_file = os.path.join(baseDir, "ref/test_polynomial.ref")
+            with BaseRegTest(ref_file, train=train) as handler:
+                s = sampling.polynomial(0, 1, 100)
+                handler.root_add_val("test_polynomial - Sample from Polynomial:", s, tol=1e-10)
+
+The key difference from a typical unit test is the optional ``train`` flag.
+When this flag is false then :meth:`baseclasses.BaseRegTest.BaseRegTest.root_add_val` will check the reference file for the value associated with the given key and compare it to the array ``s``.
```

Comment:
This method name is ridiculously long, is there a way to just have `BaseRegTest.root_add_val` and still get it to link properly, or is this just common practice?

Reply:
Yeah, you can do ``:meth:`root_add_val <baseclasses.BaseRegTest.BaseRegTest.root_add_val>` `` so the display text is shorter.